pkg/splunk/enterprise/clustermaster.go

```diff
@@ -203,6 +203,12 @@ func ApplyClusterManager(ctx context.Context, client splcommon.ControllerClient,
 			}
 		}
 	}
+	// RequeueAfter if greater than 0, tells the Controller to requeue the reconcile key after the Duration.
+	// Implies that Requeue is true, there is no need to set Requeue to true at the same time as RequeueAfter.
+	if !result.Requeue {
```

Comment:
How was the requeueAfter playing a part in the issue described?

Reply:
Thanks @vivekr-splunk . Under what circumstances did you observe the k8s controller _bug_? Do you mind providing the usecase and the impact of the bug?

Would it be better to reset `RequeueAfter` at the same place where `Requeue` is being set to false? For instance after line 202 in this file?

Reply:
well, yes if we do not set RequeueAfter in the first place and only add when its needed then, the code becomes simple. but for now, I just wanted to make sure that RequeueAfter is set to 0. if not as I mentioned it will reconcile back. I have observed this many times and discussed this with Subba. I had this code long time back. I removed it later thinking I can bring these changes later. but now this has become an issue, its difficult to read logs as logs get rotated with unnecessary controller loops and messages.